src/main/java/net/mcreator/plugin/PluginLoader.java

```diff
@@ -88,6 +90,21 @@ public PluginLoader() {
 				}
 
 				plugin.loaded = true;
+
+				if(MCreatorApplication.isInternet) {
+					if (plugin.getInfo().getUpdateJsonUrl() != null) {
+						if(!plugin.getInfo().getVersion().equals("not specified")) {
+							JsonObject version = new GsonBuilder().create()

```

Comment:
Also here instead of inline parsing, use GSON to POJO (deserilization to object)

Reply:
I don't know if it was what you meant, but I changed the GsonBuild for a normal Gson.